### DISCUSSION TITLE
Why is `response_class` necessary (instead of an annotation)

### DISCUSSION THREAD

**Discussion by septatrix** [fastapi/fastapi#4114] [Questions] [answered]
Why does `@app.get("/", response_class=HTMLResponse) async def index(): ...` work but not `@app.get("/") async def index() -> HTMLResponse: ...`? The latter seems way more natural to use

**Comment by ghandic** [comment, marked as answer]
To achieve that, you can use https://fastapi-utils.davidmontague.xyz/user-guide/inferring-router/

**Reply by septatrix** [reply]
Thanks for the link. Also the explanation on that page helped me understand the reason behind this behaviour. I tend to be rather strict with typing and explicitly cast the value before returning so I never thought about returning something different from the `response_class` type.

For now I will probably use that library although I think that it would be more intuitive to get the `response_class` from the annotations if an explicit return type (unequal to `Any`) is given.
